### PR TITLE
Fix revertTransaction hook for submitSidechainCrossChainUpdate transaction

### LINKED ISSUE #1625: Fix revertTransaction hook for submitSidechainCrossChainUpdate transaction
### Actual behavior

The revertTransaction hook for submitSidechainCrossChainUpdate transaction sets the `lastUpdated` and `lastCertificateHeight` values from the deleted block.

### Expected behavior

These values must be set from the previous blockHeader containing the `submitSidechainCrossChainUpdate` transaction from the same chainID.

### Steps to reproduce

- Check the implementation [here](https://github.com/LiskHQ/lisk-service/blob/v0.7.0-beta.0/services/blockchain-indexer/shared/indexer/transactionProcessor/interoperability/submitSidechainCrossChainUpdate.js#L71-L72)

### Which version(s) does this affect? (Environment, OS, etc...)

v0.7.0-beta.0

### PR DESCRIPTION
**Pull request by nagdahimanshu**
### What was the problem?
This PR resolves #1625 

### How was it solved?
- [x] Update revertTransaction hook implementation to consider previous block height and timestamp
- [x] Add CCU schema
- [x] Index CCU info to CCU table
- [x] Add job to delete entries from CCU until the finalised block height

### How was it tested?
Local

### CODE REVIEW

File: services/blockchain-indexer/shared/database/schema/ccu.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+module.exports = {
+	tableName: 'ccu',
+	primaryKey: 'id',
+	schema: {
+		id: { type: 'string', null: false },
```
Comment (sameersubudhi) [inline services/blockchain-indexer/shared/database/schema/ccu.js:20]:
```suggestion
	primaryKey: 'transactionID',
	schema: {
		transactionID: { type: 'string', null: false },
```

File: services/blockchain-indexer/jobs/indexer/deleteCCU.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+/*
+ * LiskHQ/lisk-service
+ * Copyright © 2022 Lisk Foundation
```
Comment (sameersubudhi) [inline services/blockchain-indexer/jobs/indexer/deleteCCU.js:3]:
```suggestion
 * Copyright © 2023 Lisk Foundation
```

File: services/blockchain-indexer/jobs/indexer/deleteCCU.js
Comment (sameersubudhi) [inline services/blockchain-indexer/jobs/indexer/deleteCCU.js:1]: I'd rename the file to `deleteFinalisedCCUMetadata.js` or something similar. Also, please rename the config, job and methods accordingly. The name `deleteCCU` can be quite confusing/misleading for users.

File: services/blockchain-indexer/shared/indexer/transactionProcessor/interoperability/submitSidechainCrossChainUpdate.js [hunk 3]
```diff
@@ -33,13 +36,27 @@ const getBlockchainAppsTable = () => getTableInstance(
...
+	ccuTableSchema.tableName,
+	ccuTableSchema,
+	MYSQL_ENDPOINT,
+);
+
+const getTransactionsTable = () => getTableInstance(
+	transactionsTableSchema.tableName,
+	transactionsTableSchema,
+	MYSQL_ENDPOINT,
+);
```
Comment (sameersubudhi) [inline services/blockchain-indexer/shared/indexer/transactionProcessor/interoperability/submitSidechainCrossChainUpdate.js:49]: With the latest mysql updates, we don't need to specify the table name anymore.
```suggestion
const getCCUTable = () => getTableInstance(ccuTableSchema, MYSQL_ENDPOINT);
const getTransactionsTable = () => getTableInstance(transactionsTableSchema, MYSQL_ENDPOINT);
```

File: services/blockchain-indexer/shared/jobs/deleteCCU.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+const config = require('../../config');
+
+const MYSQL_ENDPOINT = config.endpoints.mysql;
+const ccuTableSchema = require('../database/schema/ccu');
+
+const getCCUTable = () => getTableInstance(
+	ccuTableSchema.tableName,
+	ccuTableSchema,
+	MYSQL_ENDPOINT,
+);
```
Comment (sameersubudhi) [inline services/blockchain-indexer/shared/jobs/deleteCCU.js:28]:
```suggestion
const getCCUTable = () => getTableInstance(ccuTableSchema, MYSQL_ENDPOINT);
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]
Looks good overall but please consider the following.

File: services/blockchain-indexer/shared/indexer/transactionProcessor/interoperability/submitSidechainCrossChainUpdate.js [hunk 5]
```diff
@@ -48,22 +55,42 @@ const applyTransaction = async (blockHeader, tx, events, dbTrx) => {
...
+	const searchParams = {
+		sendingChainID: tx.params.sendingChainID,
+		propBetweens: [{
+			property: 'height',
+			lowerThan: blockHeader.height,
+		}],
+	};
+
+	const resultSet = await ccuTable.find(searchParams, 'height');
+	const prevTransactionHeight = Math.max(...resultSet.map(trx => trx.height));
```
[the remark this answers is not in the document]
Comment (nagdahimanshu) [reply services/blockchain-indexer/shared/indexer/transactionProcessor/interoperability/submitSidechainCrossChainUpdate.js]: Done! Thanks for the suggestion 👍

**Review by priojeetpriyom — APPROVED** [review]

File: services/blockchain-indexer/shared/indexer/transactionProcessor/interoperability/submitSidechainCrossChainUpdate.js [hunk 6]
```diff
@@ -48,22 +55,44 @@ const applyTransaction = async (blockHeader, tx, events, dbTrx) => {
 	};
 
 	await blockchainAppsTable.upsert(appInfo, dbTrx);
+	await ccuTable.upsert({ ...tx, ...tx.params, transactionID: tx.id }, dbTrx);
```
Comment (vardan10) [inline services/blockchain-indexer/shared/indexer/transactionProcessor/interoperability/submitSidechainCrossChainUpdate.js:58]: Should we specify the params for better readability? 
```suggestion
	await ccuTable.upsert({ height: tx.height, sendingChainID: tx.params.sendingChainID, transactionID: tx.id }, dbTrx);
```

**Review by vardan10 — APPROVED** [review]

File: services/blockchain-indexer/shared/indexer/transactionProcessor/interoperability/submitSidechainCrossChainUpdate.js [hunk 7]
```diff
@@ -48,22 +55,48 @@ const applyTransaction = async (blockHeader, tx, events, dbTrx) => {
...
+		sort: 'height:desc',
+		limit: 1,
+	};
+
+	const [{ height: prevTransactionHeight }] = await ccuTable.find(searchParams, 'height');
+	const [prevTransaction] = await transactionsTable.find(
+		{
+			height: prevTransactionHeight,
+			moduleCommand: tx.moduleCommand,
+			limit: 1,
```
Comment (sameersubudhi) [inline services/blockchain-indexer/shared/indexer/transactionProcessor/interoperability/submitSidechainCrossChainUpdate.js:91]: On the mainchain, a single block can contain multiple `.submitSideChainCrossChainUpdate` transactions from various `sendingChainIDs`. Also, we need to ensure that the relevant `prevTx.executionStatus` was success else fetch even older relevant transaction.

Ideally with `if (tx.executionStatus !== TRANSACTION_STATUS.SUCCESS) return;` within the hooks, the above shouldn't be an issue but it's better to cover any possible failure scenarios to make the code robust.

**Review by sameersubudhi — CHANGES_REQUESTED** [review]

**Review by sameersubudhi — APPROVED** [review]

**Pull request merged**